design/thin-edge-core.md

```diff
@@ -4,24 +4,31 @@ A typical IoT application has to address:
+* On one side, to address the diversity of use-cases and integration requirements, one needs a large open set of features that can be easily extended and combined.
   [NodeRed](https://nodered.org/) is the archetype of such tools facilitating connections of components.
 * On the other side, to meet resource and security constraints, one needs a minimal system specifically built for a task with cherry-picked components.
-  On the extreme side this leads to the idea of [unikernel](https://en.wikipedia.org/wiki/Unikernel) with sealed, single-purpose software images.
+  On the extreme side, this leads to the idea of [unikernel](https://en.wikipedia.org/wiki/Unikernel) with sealed, single-purpose software images.
   
-Can these two poles be reconciled?
+What makes thin-edge unique is its approach to reconcile these two poles with two levels of building blocks.
+* To maximize flexibility and interoperability, thin-edge provides the tools to build an agent from independent executables that interact over MQTT using JSON messages.
+* To minimize resources and vulnerabilities, thin-edge features fine-grain Rust components as well as the rules to combine them into purpose-specific agents.
+* To make feasible the assemblage of fine-grain and MQTT-based components into IIoT agents,
+  thin-edge comes with an extensible data model for IIoT - cloud connectivity, telemetry data, device management, child devices, ... -
+  and defines the rules to exchange related messages over MQTT as well as between fine-grain components and at the boundary between the internal and MQTT worlds.
 
-The approach of thin-edge is to ease the development of IoT applications on edge devices with a smooth transition between:
-* prototyping use-cases - when the aim is to easily onboard devices for IIoT,
-* production use-cases - when the need is to deploy a hardened specific software on a large fleet of devices,
-* as well as intermediate use-cases - when the application is build by several vendors with components written using different programming languages;
-  or when a manufacturer assembles his tools on the devices while letting open to his customers the option to add their own modules. 
+With this approach, thin-edge eases the development of IoT applications on edge devices with a smooth transition between:
+* prototyping use-cases - when the need is to easily onboard devices for IIoT with batteries-included tools,
+* production use-cases - when the need is to deploy a hardened specific software on a large fleet of devices,  
+* IIoT agent development - when the aim is to implement an application-specific agent that integrates tools and services provided by independent vendors.
+* Hardware specific software - when a manufacturer assembles his tools on the devices while letting open to his customers the option to add their own modules. 
```

Comment:
I am not sure if Hardware specific software is the right word. How about App development?

Reply:
This whole paragraph has been rewritten - because confusing for many.